url/resources/urltestdata.json

```diff
@@ -7794,5 +7794,204 @@
     "protocol": "wss:",
     "search": "",
     "username": ""
+  },
+  "Forbidden host codepoints",
+  { "input": "foo://ho\u0000st/",
```

Comment:
Would it be possible to add these tests near the tests added in #23572?

Also, the prevailing style in the file is to have `{` not be on the same line as the first property.